docroot/modules/custom/mass_admin_toolbar/src/Plugin/Block/MessageTabs.php

```diff
@@ -64,9 +64,12 @@ public function build() {
     $cacheability = new CacheableMetadata();
     $primary = $this->localTaskManager->getLocalTasks($this->routeMatch->getRouteName(), 0);
     $secondary = $this->localTaskManager->getLocalTasks($this->routeMatch->getRouteName(), 1);
+    $local_tasks = CacheableMetadata::createFromObject($this->localTaskManager);
+
     $cacheability = $cacheability
       ->merge($primary['cacheability'])
-      ->merge($secondary['cacheability']);
+      ->merge($secondary['cacheability'])
+      ->merge($local_tasks);
```

Comment:
Nice find ... I think you can replace these lines with (untested):

```
$cacheability
  ->addCacheableDependency($primary)
  ->addCacheableDependency($secondary)
  ->addCacheableDependency($this->localtaskManager);
```

Reply:
This is the one with the suggested change
<img width="1142" alt="Screenshot 2024-05-15 at 18 10 19" src="https://github.com/massgov/openmass/assets/62405127/30e2d54e-7164-4157-bc07-a41628c9f486">
This is the one with the change suggested in the PR. as you can see if we change this we will lose some cache tags and the route cache context
<img width="1356" alt="Screenshot 2024-05-15 at 18 10 01" src="https://github.com/massgov/openmass/assets/62405127/0c31bfd6-baf0-4e22-a5da-42875eb8ef4b">